overlord/ifacestate/handlers.go

```diff
@@ -227,6 +227,24 @@ func (m *InterfaceManager) doConnect(task *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// connect OS snap if snap name is omitted
+	if slotRef.Snap == "" {
+		slotRef.Snap = "ubuntu-core"
+	}
+
+	// connect the plug to slot with matching interface if slot is omitted
+	if slotRef.Name == "" {
```

Comment:
Not sure how hard this is to do, but a unit test for this would be nice (quicker turnaround than spread :)